### PR TITLE
Add loading state for routes.

### PR DESCRIPTION
**Pull request by yuchenshi**
Screen Shot 2020-04-01 at 11 12 16 AM

### CODE REVIEW

File: src/components/Firestore/index.tsx [hunk 1]
```diff
@@ -147,9 +142,17 @@ export const Firestore: React.FC<FirestoreProps> = ({ config, projectId }) => {
...
+export const FirestoreLoading: React.FC = () => (
+  <GridCell
+    span={12}
+    className="Firestore-loading"
+    data-testid="firestore-loading"
+  >
+    <CircularProgress size="xlarge" />
+    <Typography use="body2" tag="p">
+      Firestore Emulator Loading...
+    </Typography>
```
Comment (tstirrat) [inline src/components/Firestore/index.tsx:154]: This and the centering styles are reused, can you extract into a component?

In console we have `<fire-spinner message="Blah loading" />`, where the default is to show a spinner only. or include the message if provided.

I think something like that would be helpful.

Comment (tstirrat) [reply src/components/Firestore/index.tsx:154]: And we could use without the grid and centering stuff with `<Spinner inline size="xsmall" />`

Perhaps you could even include the Grid and span={}.. its often going to be included as a grid flow: `<Spinner span={12} />`

Comment (tstirrat) [reply src/components/Firestore/index.tsx:154]: The extra component stuff is easy to do later, but making it into a simple component `<Spinner message="test" />` is helpful now.

Comment (yuchenshi) [reply src/components/Firestore/index.tsx:157]: Added the grid version now but I'll leave the `inline` version unimplemented until we actually need it.

File: src/components/Firestore/index.tsx [hunk 2]
```diff
@@ -147,9 +143,17 @@ export const Firestore: React.FC<FirestoreProps> = ({ config, projectId }) => {
   );
 };
 
-export const FirestoreRouteLoading: React.FC = () => (
-  // TODO
-  <div>Loading...</div>
+export const FirestoreLoading: React.FC = () => (
```
Comment (tstirrat) [inline src/components/Firestore/index.tsx:146]: is this dead code now?

**Pull request merged**